### DISCUSSION TITLE
Deployment Issue - redirect to localhost:5000

### DISCUSSION THREAD

**Discussion by CartyBoston** [assemblee-virtuelle/activitypods#248] [General]
Team,

I can bring up the backend and frontend locally using the instructions in the README. I knew little about JavaScript and nothing about yarn, node, and docker. It just worked, thank you.

I have learned some about the new (to me) tools and I'd like to make my pod server available to the outside world.

I found a couple of issues in /deploy:

1. React seems to want node 18 now, the backend and frontend /deploy Docker files build on node 14.

    149.3 error react-hook-form@7.49.2: The engine "node" is incompatible with this module. Expected version ">=18". Got “14.21.3"

I updated to:

 FROM node:18-alpine 

and am able to build but I don't know if that creates other issues.

2. In the /deploy docker-compose.yml the arena section contains:

  build: ./arena/Docker

which upsets for some reason (maybe a bug) my version of docker:

  [+] Building 0.0s (0/0)                                                                                   docker:default
  unable to prepare context: path "/home/carty/activitypods/deploy/arena/Dockerfile" not found
  
  I removed the "/Dockerfile" like the other sections and it built.

3. When I bring up all the containers and use the external address for the server - https://pods.roundpond.net - something is  immediately redirecting me to "localhost:5000" which of course does not work.

I wonder if I got something wrong in deploy/docker-compose or the .env-locals.

Can you help me think about how to debug? I want to make my friends use my server :).

**Comment by Laurin-W** [comment]
Hey Carty,

regarding the __(1) node version__ -- I'm mostly using node 20 at the moment and it's working fine; node 18 should work too. I'm usually not working in a docker environment, and use  `nvm` for node version control. In case you need it, you can install by running `npm install -g nvm` and then call `nvm use 18`.
As to why the docker container is still configured this way -- @srosset81 we should probably update this?

Regarding __(2) arena docker issues__, that sounds strange. I haven't worked with that before though. But @srosset81 might have an idea :)

Regarding __(3) redirect to `localhost:5000`__ --- this ist the default frontend address. You can set your frontend URI in your backend's `.env` file (the default one is located at `/backend/.env` as a reference). Seting `SEMAPPS_FRONTEND_URL` to your frontend address in the `.env.local` file should do the trick. E.g. for our `https://mypod.store/` this is `https://app.mypod.store/`.

Hope that helps! :)

**Comment by CartyBoston** [comment]
Lauren thanks,

I kind of had a "no docker" philosophy because I wanted to understand how to manage my services, but it's so pervasive and I was really impressed with bringing up e.g fuseki with a single command, I see the appeal, I will use it carefully, like a chainsaw.

You know what would be super helpful for people looking to deploy? A page that describes the components:

 - reverse web proxy (traefik) -> triplestore (fuseki) -> redis (cache) -> arena -> backend -> frontend

and in particular the config (high level, ports and such) of the interface between them. Like I was unfamiliar with traefik but have used nginx to reverse proxy, it'd be nice to swap that in.

As soon as I get my instance up I'd be happy to draft such a doc but I am brand new and have zero ability to prioritize, I need your sincere feedback.

If @srosset81 is OK with any updates to the /deploy docker files I'd suggest you get me to do it, I am dying to be helpful. Shall I do a pull request? (note: I know enough about git to know a "pull request" is a thing, but I don't have a ton of experience coding collaboratively, learning)

**Reply by Laurin-W** [reply]
> You know what would be super helpful for people looking to deploy? A page that describes the components:
    reverse web proxy (traefik) -> triplestore (fuseki) -> redis (cache) -> arena -> backend -> frontend

That's a great idea! We have had a couple of people doing deployments with different levels of expertise and talked about this as an action item, to improve deployment experience and documentation. I imagine it could be quite helpful to have someone write down their fresh experience on this. If you'd like to create a pull request, this would be great :)

Same for the docker files but I didn't author them so asking @srosset81 is a good idea :)

**Reply by srosset81** [reply]
Hi @CartyBoston !
Thanks for the messages. I would recommend to have a look at the new deployment guide at https://docs.activitypods.org/tutorials/deploy-your-own-pod-provider/ It is much better, and it will help you have everything up and running in a few minutes.
If you have suggestions on how to improve this guide (for example with a better description of the services), feel free to open a PR !
